### PR TITLE
[Sol->Yul] Implementing member access to arrays in calldata structs.

### LINKED ISSUE #9767: [Sol->Yul] Implement member access for arrays in calldata structs
Member access for calldata structs is implemented. However, if struct contains arrays, member access is failing for these.

example:

```
contract C {
    struct S {
        uint128[3] a;
    }

    S[] s;

    function f(S[] calldata c) external {
        s = c;
        assert(s[1].a[0] == c[1].a[0]);
    }
}
```

Fails as `c[1].a[0]` fails because member access doesn't support arrays

### PR DESCRIPTION
**Pull request by mijovic**
Fixes #9767

### CODE REVIEW

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp [hunk 1]
```diff
@@ -1810,6 +1810,8 @@ void IRGeneratorForStatements::endVisit(MemberAccess const& _memberAccess)
 					", " <<
 					offset <<
 					")\n";
+			else if (dynamic_cast<ArrayType const*>(_memberAccess.annotation().type))
```
Comment (chriseth) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp]: What about structs?

Comment (mijovic) [reply libsolidity/codegen/ir/IRGeneratorForStatements.cpp]: I'll add them as well now. It should be the same case.

Comment (mijovic) [reply libsolidity/codegen/ir/IRGeneratorForStatements.cpp]: Fixed for structs as well. Also added test to check functionality.

**Review by chriseth — DISMISSED** [review]

**Comment by mijovic** [comment]
@chriseth Just after you approved I pushed a change (activated one more test via yul).

**Review by chriseth — APPROVED** [review]

**Pull request merged**